cmd/lifecycle/main.go

```diff
@@ -18,31 +18,41 @@ import (
 )
 
 func main() {
-	switch strings.TrimSuffix(filepath.Base(os.Args[0]), filepath.Ext(os.Args[0])) {
+	phase := strings.TrimSuffix(filepath.Base(os.Args[0]), filepath.Ext(os.Args[0]))
+	if phase == "rebaser" {
+		cli.Run(&rebaseCmd{Platform: platform.NewPlatformFor(platformAPIWithExitOnError(), "<layers>")}, false)
```

Comment:
it's less clear why this one case needs to be above the switch though?  it seems like this is the only one that doesn't call `cli.FlagLayersDir` , along with the Exit / ExitWithVersion cases?  it looked like even everything under `subcommand` calls that -- were you trying to factor it out so we only have one line that says `cli.FlagLayersDir(&layersDir)` ?

Reply:
Yeah, when you call `cli.FlagLayersDir(&layersDir)` that registers the flag as valid, and it isn't for the rebaser. Alternatively we could do something like
```
if phase == "rebaser" {
		cli.Run(&rebaseCmd{Platform: platform.NewPlatformFor(platformAPIWithExitOnError(), "<layers>")}, false) 
} else {
		cli.FlagLayersDir(&layersDir) // assumes all phases that follow (including invalid phases) respect -layers
}
```
Would that be clearer? I'm open to other suggestions too

Reply:
if it really is valid for every single other thing (which it seemed like?) then your if/else technology above would be a good solution.   otherwise we'll just have to live with it being _almost_ everywhere.  nbd

Reply:
Sounds good. Let's do if/else, it's clearer and less duplicative. I'll make the change